src/maestro/agents/dspy_agent.py

```diff
@@ -82,20 +82,28 @@ async def run(self, prompt: str, context=None, step_index=None) -> str:
             dspy_tools = []
             if self.tool_names and len(self.tool_names):
                 for tool_name in self.tool_names:
+                    # print(f"tool: {tool_name}")
                     dspy_tools.extend(
                         await get_mcp_tools(
                             tool_name, dspy.Tool.from_mcp_tool, mcp_stack
                         )
                     )
             self.print(f"Running Dspy agent: {self.agent_name} with prompt: {prompt}\n")
             self.dspy_agent = dspy.ReAct(self.dspy_signature, dspy_tools)
+            result = {}
             try:
                 result = await self.dspy_agent.acall(user_request=prompt)
             except Exception as e:
                 print(f"Agent error: {e}")
-            self.print(f"Response from {self.agent_name}: {result.process_result}\n")
+
             await mcp_stack.aclose()
-            return result.process_result
+            if result and result.process_result:
+                self.print(
+                    f"Response from {self.agent_name}: {result.process_result}\n"
+                )
+                return result.process_result
+            print("No response from Agent")
+            return "Agent error"
```

Comment:
should we really just return this string instead of an error? I think a better solution would be to raise an error clarifying that there was no response from the agent

Reply:
We don't know what happens in the DSPy agent.  It just doesn't return response.  So it is returning this message or raising an exception is another way

Reply:
I think raising an exception is the better path, because if the agent doesn't respond that's not a valid case. The issue this PR is trying to solve is around better error messaging, but this would print `Agent error` to the log, but then continue on as if the agent returned the response "Agent error"